libraries/app/api.cpp

```diff
@@ -550,4 +560,82 @@ namespace graphene { namespace app {
        return fc::ecc::range_get_info( proof );
     }
 
+    // asset_api
+    asset_api::asset_api(graphene::chain::database& db) : _db(db) { }
+    asset_api::~asset_api() { }
+
+    vector<account_asset_balance> asset_api::get_asset_holders( asset_id_type asset_id ) const {
+
+      const auto& bal_idx = _db.get_index_type< account_balance_index >().indices().get< by_asset_balance >();
+      auto range = bal_idx.equal_range( boost::make_tuple( asset_id ) );
+
+      vector<account_asset_balance> result;
+
+      for( const account_balance_object& bal : boost::make_iterator_range( range.first, range.second ) )
+      {
+        if( bal.balance.value == 0 ) continue;
+
+        auto account = _db.find(bal.owner);
+
+        account_asset_balance aab;
+        aab.name       = account->name;
+        aab.account_id = account->id;
+        aab.amount     = bal.balance.value;
+
+        result.push_back(aab);
+      }
+
+      return result;
+    }
+	
+	// get number of asset holders.
+   int asset_api::get_asset_holders_count( asset_id_type asset_id ) const {
+
+      const auto& bal_idx = _db.get_index_type< account_balance_index >().indices().get< by_asset_balance >();
+      auto range = bal_idx.equal_range( boost::make_tuple( asset_id ) );
```

Comment:
If there's not a `count` method, there is `std::distance(range.begin(), range.end())` ([docs](http://en.cppreference.com/w/cpp/iterator/distance)) which is also fewer lines and potentially faster (`distance` selects the algorithm based on the capabilities of the iterators).